src/modules/load_mon/LoadMon.cpp

```diff
@@ -73,15 +73,30 @@ int LoadMon::task_spawn(int argc, char *argv[])
 	_object.store(obj);
 	_task_id = task_id_is_work_queue;
 
-	/* Schedule a cycle to start things. */
-	obj->start();
+	if (obj->start() == PX4_OK) {
+		return PX4_OK;
+	}
 
-	return 0;
+	delete obj;
+	_object.store(nullptr);
+	_task_id = -1;
+
+	return PX4_ERROR;
 }
 
-void LoadMon::start()
+int LoadMon::start()
 {
+#if defined(__PX4_LINUX)
+	_proc_fd = fopen("/proc/meminfo", "r");
```

Comment:
One last little point, would you mind moving this to open in `Run()`? It doesn't currently matter on Linux because these are all are threads in the same px4 process, but on NuttX you get in trouble with this pattern because `start()` is called from a separate task (files/streams are stored per task). So as a general rule I think it would be best if we skip all resource acquisition in `start()` like this before someone accidentally duplicates it somewhere else on NuttX.

Reply:
Done